tests/unit/test_read_configs_include.py

```diff
@@ -547,6 +567,156 @@ def test_allowed_no_override_old_variables_when_set(self, read_config_include_fi
         print(proc.stdout)
         assert split_and_strip(get_command_stdout(proc))[-1] == "pavics.example.com"
 
+    @pytest.mark.parametrize("from_name,to_name",
+        (("ENABLE_JUPYTERHUB_MULTI_NOTEBOOKS", "JUPYTERHUB_ENABLE_MULTI_NOTEBOOKS"),
+         ("JUPYTERHUB_ENABLE_MULTI_NOTEBOOKS", "ENABLE_JUPYTERHUB_MULTI_NOTEBOOKS")))
+    def test_formatting_preserved_from_old_to_new_var_vice_versa(self, read_config_include_file,
+                                                                 exit_on_error, from_name, to_name):
+        """
+        Test that formatting (new lines, leading spaces, quotes) are preserved when old var
+        value is transfered to new var and vice-versa.  This is important during template expansion.
+        """
+        expected = ("\n"
+                    "    # python code requires keeping formatting  \n"
+                    "    {'user': 'pass'}\n")
+
+        extra = {
+            from_name: f"\"{expected}\"",
+            "BIRDHOUSE_BACKWARD_COMPATIBLE_ALLOWED": "True",
+        }
+        proc = self.run_func(
+            read_config_include_file,
+            extra,
+            f'echo "${to_name}"',
+            exit_on_error=exit_on_error,
+        )
+        print(proc.stdout)
+        assert "\n".join(get_command_stdout(proc).split("\n")[-4:]) == expected
+
+    def test_template_expansion_enabled_for_old_var(self, read_config_include_file, exit_on_error):
+        """
+        Test that template expansion is enabled for corresponding old var if new var is enabled.
+        """
+
+        env_local = '''
```

Comment:
I would prefer to either make these global constants so that the indentation within the function doesn't get all confusing. If you want to keep them in the function itself then please use something like:

```python
        env_local = "\n".join([
            "",  
            "# Add custom backward compatible var mapping.",
            "BIRDHOUSE_BACKWARDS_COMPATIBLE_VARIABLES=\"$BIRDHOUSE_BACKWARDS_COMPATIBLE_VARIABLES\"",
            "    MY_OLD_VAR=MY_NEW_VAR"",
            "",
            ...
        ])
```

Reply:
Or, use test files, and employ actual formatting without misinterpretation/mangling by strings.

Reply:
These are specific config for each test, why make them global constant?

And the syntax I use the equivalent of "here doc" in Python to make multi lines string as simple as possible.  Why surround with "\n".join() if we don't need to.  Using here-doc, the will be less "misinterpretation/mangling by strings".

I'd rather choose simplicity over strict indentation and possibility of error-prone.

Reply:
> why make them global constant?

Make them a fixture then.

Python code that is suddenly de-indented mid function is hard to read and makes it difficult to easily determine the nesting hierarchy.

I agree with @fmigneault that using a test file would work well for this too (we already have some in the `fixtures` directory ... feel free to add more).